basic_hashtable/b_hashtables.c

```diff
@@ -84,7 +89,17 @@ BasicHashTable *create_hash_table(int capacity)
  ****/
 void hash_table_insert(BasicHashTable *ht, char *key, char *value)
 {
-
+  int h = hash(key, ht->capacity);
+  if (ht->storage[h] != NULL)
+  {
+    printf("\nOverwriting existing value\n");
+    destroy_pair(ht->storage[h]);
+    ht->storage[h] = create_pair(key, value);
```

Comment:
C code is all about being efficient. You have `ht->storage[h] = create_pair(key, value);` twice
Let's clean it up with
```
  if (ht->storage[h] != NULL)
  {
    printf("\nOverwriting existing value\n");
    destroy_pair(ht->storage[h]);
  }
  ht->storage[h] = create_pair(key, value);
```